storage-node/src/services/sync/acceptPendingObjects.ts

```diff
@@ -0,0 +1,212 @@
+
+    pendingDataObjects.forEach((dataObject) => {
+      const storageBucket = dataObject.storageBag.storageBuckets.find(({ id }) => this.uploadBuckets.includes(id))
+      if (storageBucket) {
+        this.push(dataObject.id, storageBucket.id, dataObject.storageBag.id)
+      } else {
+        logger.warn(
+          `Data object ${dataObject.id} in pending directory is not assigned to any of the upload buckets: ${this.uploadBuckets}.`
+        )
+      }
+    })
+  }
+
+  private async loadPendingDataObjects(): Promise<void> {
+    const pendingIds = await this.getPendingObjectsFromLocalDir(this.pendingDataObjectsDir)
+
+    await this.loadPendingDataObjectsFromIDs(pendingIds)
+
+    logger.debug(`Pending data objects ID cache loaded.`)
+  }
+
+  public push(dataObjectId: string, storageBucketId: string, bagId: string): void {
+    this.pendingDataObjects.set(dataObjectId, [storageBucketId, bagId])
+  }
+
+  private async popN(n: number): Promise<AcceptPendingDataObjectsParams[]> {
+    const params: AcceptPendingDataObjectsParams[] = []
+    let count = 0
+
+    while (count < n && this.pendingDataObjects.size > 0) {
+      // Extract the first element from the map
+      const [dataObjectId, [storageBucketId, bagId]]: [string, [string, string]] = this.pendingDataObjects
+        .entries()
+        .next().value
+      this.pendingDataObjects.delete(dataObjectId)
+
+      // Find or create the storage bucket in the params array
+      let storageBucket = params.find((p) => p.storageBucket.id === storageBucketId)
+      if (!storageBucket) {
+        const account = this.bucketKeyPairs.get(storageBucketId.toString())
+        if (!account) {
+          logger.error(`No key pair found for storage bucket ${storageBucketId}.`)
+          continue
+        }
+
+        storageBucket = {
+          account,
+          storageBucket: {
+            id: storageBucketId,
+            bags: [],
+          },
+        }
+        params.push(storageBucket)
+      }
+
+      // Find or create the bag in the storage bucket in the params array
+      let bag = storageBucket.storageBucket.bags.find((b) => b.id === bagId)
+      if (!bag) {
+        bag = {
+          id: bagId,
+          dataObjects: [],
+        }
+        storageBucket.storageBucket.bags.push(bag)
+      }
+
+      // Add the data object to the bag in the params array
+      bag.dataObjects.push(dataObjectId)
+
+      count++
+    }
+
+    return params
+  }
+
+  /**
+   * Returns file names from the pending objects directory.
+   *
+   * @param directory - local directory to get file names from
+   */
+  private async getPendingObjectsFromLocalDir(directory: string): Promise<string[]> {
+    // Check if directory exists and if not, create it
+    if (!fs.existsSync(directory)) {
+      await fsPromises.mkdir(directory)
+    }
+
+    // Read the directory contents
+    return await fsPromises.readdir(directory)
```

Comment:
I have an issue with both the temp dir and now the pending dir being sub directories of the uploads dir.

But that aside (I'll make a separate issue for it), we should make sure the sync worker which also loads objects with `readdir` ignores the pending dir here https://github.com/Joystream/joystream/blob/9c5383d213667d750d9150d4483982cab84ff0e4/storage-node/src/services/caching/localDataObjects.ts#L34